File: src/main/java/net/siegemc/core/items/items/TestWand.java
package net.siegemc.core.items.items;

import net.siegemc.core.Core;
import net.siegemc.core.items.Rarity;
import net.siegemc.core.items.Stat;
import net.siegemc.core.items.WandItem;
import org.bukkit.Material;
import org.bukkit.Particle;
import org.bukkit.inventory.ItemStack;

import java.util.Arrays;

public class TestWand extends WandItem {
    public TestWand() {
        super("Test Wand", new ItemStack(Material.STICK));
        this.setManaRequired(10);
        this.setParticle(Particle.COMPOSTER);
        this.setDamage(10000);
        this.setDamageRadius(5.5);
        this.setRange(120);
        this.setRarity(Rarity.DEBUG);
        this.getStats().put(Stat.STRENGTH, 100);
        this.setDescription(Arrays.asList("This is just a test wand used", "for testing wands"));
        Core.getItems().put("Test Wand", this);
    }
}

File: src/main/kotlin/net/siegemc/core/items/implemented/equipment/wands/TestWand.kt
package net.siegemc.core.items.implemented.equipment.wands

import net.siegemc.core.items.CustomItemUtils.statMap
import net.siegemc.core.items.implemented.equipment.armor.chestplates.TestChestplate
import net.siegemc.core.items.implemented.materials.TestMaterial
import net.siegemc.core.items.recipes.CustomRecipe
import net.siegemc.core.items.types.equipment.CustomWand
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class TestWand @Deprecated("Specify quality") constructor() : CustomWand(
    name = "Test Wand",
    customModelData = 1,
    description = listOf("A wand for testing"),
    levelRequirement = 0,
    material = Material.STICK,
    recipe = CustomRecipe(listOf(null, null, TestMaterial(0, 1), TestMaterial(0, 1), null, null, null, null), true) {
        return@CustomRecipe TestWand((0..100).random())
    },
    baseStats = statMap(strength = 10.0),
    range = 15,
    red = 100,
    green = 100,
    blue = 100,
    manaRequired = 2,
    damageRadius = 3.5
) {

    @Suppress("DEPRECATION")
    constructor(item: ItemStack): this() {
        this.item = item
        deserialize()
    }

    @Suppress("DEPRECATION")
    constructor(quality: Int): this() {
        this.quality = quality
    }

    init {
        serialize()
        updateMeta()
    }

}